src/cli/self_update.rs

```diff
@@ -630,7 +630,17 @@ fn install_bins() -> Result<()> {
     // NB: Even on Linux we can't just copy the new binary over the (running)
     // old binary; we must unlink it first.
     if rustup_path.exists() {
-        utils::remove_file("rustup-bin", &rustup_path)?;
+        if cfg!(windows) {
+            let removal_path = bin_path.join(&format!("rustup-to-delete{}", EXE_SUFFIX));
+            if removal_path.exists() {
+                utils::remove_file("rustup-to-delete", &removal_path)?;
+            }
+            fs::rename(&rustup_path, &removal_path).chain_err(|| {
```

Comment:
This will need special flags if the rustup_path may still be in use, but if its not in use, great.

Reply:
The approach is deliberate because `rustup_path` *is* in use at this point.  Windows permits renaming the binary so long as it doesn't get deleted things are okay.  This was the only approach I could come up with which didn't require the parent `rustup` process to exit first.